tests/bgp/test_bgp_allow_list.py

```diff
@@ -357,77 +355,17 @@ def check_other_neigh(nbrhosts, permit, node=None, results=None):
-            duthost.copy(content=yaml.safe_dump(constants), dest=CONSTANTS_FILE)
-
-            logger.info('Restart BGP for default_action permit to take effect')
-            duthost.shell('systemctl reset-failed bgp')  # Workaround for issue 'Start request repeated too quickly'
-            duthost.shell('systemctl restart bgp')
-            pytest_assert(wait_until(300, 30, self.bgp_started, duthost, tbinfo), 'Service bgp not started')
-
-        return
-
-    def test_empty_allow_list_permit(self, duthosts, rand_one_dut_hostname, setup, nbrhosts):
-        duthost = duthosts[rand_one_dut_hostname]
-        self.check_routes_on_tor1(setup, nbrhosts)
-        self.check_routes_on_dut(duthost)
-        self.check_routes_on_neighbors_empty_allow_list(nbrhosts, setup, permit=True)
-
-    def test_allow_list_permit(self, duthosts, rand_one_dut_hostname, setup, nbrhosts, load_remove_allow_list):
-        duthost = duthosts[rand_one_dut_hostname]
-        self.check_routes_on_tor1(setup, nbrhosts)
-        self.check_routes_on_dut(duthost)
-        self.check_routes_on_neighbors(nbrhosts, setup, permit=True)
-
-
-class TestBGPAllowListDeny(BGPAllowListBase):
-
-    @pytest.fixture(scope='class', autouse=True)
-    def default_action_deny(self, duthosts, rand_one_dut_hostname, tbinfo):
-        duthost=duthosts[rand_one_dut_hostname]
-        constants = yaml.safe_load(duthost.shell('cat {}'.format(CONSTANTS_FILE))['stdout'])
-        default_action = constants['constants']['bgp']['allow_list']['default_action']
-
-        if default_action != 'deny':
-            logger.info('Set constants.bgp.allow_list.default_action to deny')
-            constants['constants']['bgp']['allow_list']['default_action'] = 'deny'
-            duthost.copy(content=yaml.safe_dump(constants), dest=CONSTANTS_FILE)
-
-            logger.info('Restart BGP for default_action deny to take effect')
-            duthost.shell('systemctl reset-failed bgp')  # Workaround for issue 'Start request repeated too quickly'
-            duthost.shell('systemctl restart bgp')
-            pytest_assert(wait_until(300, 30, self.bgp_started, duthost, tbinfo), 'Service bgp not started')
-
-        yield
-
-        logger.info('Set constants.bgp.allow_list.default_action to permit')
-        constants['constants']['bgp']['allow_list']['default_action'] = 'permit'
-        duthost.copy(content=yaml.safe_dump(constants), dest=CONSTANTS_FILE)
-
-        logger.info('Restart BGP for default_action deny to take effect')
-        duthost.shell('systemctl reset-failed bgp')  # Workaround for issue 'Start request repeated too quickly'
-        duthost.shell('systemctl restart bgp')
-        wait_until(300, 20, self.bgp_started, duthost, tbinfo)
-
-    def test_empty_allow_list_deny(self, duthosts, rand_one_dut_hostname, setup, nbrhosts):
+    def test_default_allow_list(self, duthosts, rand_one_dut_hostname, setup, nbrhosts):
+        permit = True if DEFAULT_ACTION == "permit" else False
```

Comment:
The `default_action` in `/etc/sonic/constants.py` could have values 'permit' or 'deny'. This change will only cover one of the values (whatever by default in `/etc/sonic/constants.py`), not all of them. The original script can cover both. Could you elaborate why make this change?

Reply:
@wangxin Default value define in /etc/sonic/constants.yml is static value for given image. Application can not modify it. We have per prefix list default action now. So the case of updating the file and restarting bgp is not a valid use case . Reason for this change to have test case that verify the default action the system comes up (and don't update it)  before  we program allow prefix list and it's action.